src/cli/sandbox.cpp

```diff
@@ -63,6 +67,56 @@ bool Sandbox::init()
       }
 
    return (::cap_enter() == 0);
+#elif defined(BOTAN_TARGET_OS_HAS_SETPPRIV)
+   priv_set_t *ps;
+
+   if ((ps = ::priv_allocset()) == nullptr)
+      {
+      return false;
+      }
+
+   ::priv_basicset(ps);
+
+   if (::priv_delset(ps, PRIV_PROC_FORK) == -1)
+      {
+      ::priv_freeset(ps);
+      return false;
+      }
+
+   if (::priv_delset(ps, PRIV_PROC_EXEC) == -1)
+      {
+      ::priv_freeset(ps);
+      return false;
+      }
+
+   if (::priv_delset(ps, PRIV_PROC_INFO) == -1)
+      {
+      ::priv_freeset(ps);
+      return false;
+      }
+
+   if (::priv_delset(ps, PRIV_PROC_SESSION) == -1)
+      {
+      ::priv_freeset(ps);
+      return false;
+      }
+
+   if (::setppriv(PRIV_SET, PRIV_PERMITTED, ps) == -1)
+      {
+      ::priv_freeset(ps);
+      return false;
+      }
+
+   if (::setppriv(PRIV_SET, PRIV_INHERITABLE, ps) == -1)
+      {
+      ::priv_freeset(ps);
+      return false;
+      }
+
+   ::priv_emptyset(ps);
```

Comment:
Elsewhere we `freeset` without using `emptyset` - is that ok?

Reply:
might be better indeed.